src/importers/postman/Parser.js

```diff
@@ -0,0 +1,583 @@
+                    environments.push(environment)
+                }
+            }
+        }
+        /* .postman_collection */
+        else if (obj.requests) {
+            collections.push(obj)
+        }
+        /* .postman_environment */
+        else if (obj.values && obj.name) {
+            environments.push(obj)
+        }
+        else {
+            throw new Error('Invalid Postman file (missing required keys)')
+        }
+
+        this.context = this._createContext(environments, collections)
+        return this.context
+    }
+
+    // @tested
+    _createContext(environments, collections) {
+        let envs = new Immutable.List(
+            environments.map(
+                env => {
+                    return this._importEnvironment(env)
+                }
+            )
+        )
+
+        let baseGroup = collections.reduce(
+            (rootGroup, collection) => {
+                let group = this._importCollection(collection)
+                return rootGroup.setIn(
+                    [ 'children', group.get('id') ], group
+                )
+            },
+            new Group()
+        )
+
+        let context = new RequestContext({
+            environments: envs,
+            group: baseGroup
+        })
+
+        return context
+    }
+
+    // @tested
+    _importEnvironment(environment) {
+        let env = new Environment({
+            id: environment.id,
+            name: environment.name
+        })
+
+        if (environment.values) {
+            for (let value of environment.values) {
+                env = env.setIn(
+                    [ 'variables', value.key ],
+                    new KeyValue({
+                        key: value.key,
+                        value: this._referenceEnvironmentVariable(value.value),
+                        valueType: value.type || null
+                    })
+                )
+            }
+        }
+
+        return env
+    }
+
+    // @tested
+    _importCollection(collection) {
+        if (!collection.requests) {
+            throw new Error('Invalid Postman file (missing data)')
+        }
+
+        let requestsById = {}
+        for (let req of collection.requests) {
+            let request = ::this._createRequest(collection, req)
+            requestsById[req.id] = request
+        }
+
+        return this._createGroupFromCollection(
+            collection, requestsById
+        )
+    }
+
+    // @tested
+    _referenceEnvironmentVariable(string) {
+        if (typeof string === 'undefined' || string === null) {
+            return null
+        }
+
+        let groups = []
+        let stack = [
+            {
+                start: -1,
+                end: string.length,
+                depth: 0,
+                ref: new EnvironmentReference()
+            }
+        ]
+        let group
+        let level = 0
+        let prevChar = ''
+        let notEvaluatedIndex = 0
+        let i = 0
+        for (let char of string) {
```

Comment:
What's the goal of this char-by-char string parsing? Why not simply a regexp to match `{{env}}`? `/\{\{[^}]+\}\}/` would work nicely…
If this is to be able to resolve nested env variables, I'm concerned that we are trading a "nice to have" feature that is likely to not even be officially supported by Postman – and may not be even used by anyone (why on Earth would you nest environment variable names?!) – for a massive performance hit. Most JavaScript regexp engines are probably written in plain C, and is super efficient, doing this by hand here, on every single string we import, may be really slow…

Maybe should we keep this function around, as an alternative? And quickly make a simpler one (can be done in a few lines I think). It's also about code complexity, I wouldn't want to resolve bugs in this myself :)